android-components/components/feature/search/src/main/java/mozilla/components/feature/search/telemetry/BaseSearchTelemetry.kt

```diff
@@ -71,67 +72,101 @@ abstract class BaseSearchTelemetry {
-            regexp = "^https:\\/\\/www\\.bing\\.com\\/search",
-            queryParam = "q",
-            codeParam = "pc",
-            codePrefixes = listOf("MOZ", "MZ"),
-            followOnCookies = listOf(
-                SearchProviderCookie(
-                    extraCodeParam = "form",
-                    extraCodePrefixes = listOf("QBRE"),
-                    host = "www.bing.com",
-                    name = "SRCHS",
-                    codeParam = "PC",
-                    codePrefixes = listOf("MOZ", "MZ"),
-                ),
-            ),
-            extraAdServersRegexps = listOf(
-                "^https:\\/\\/www\\.bing\\.com\\/acli?c?k",
-                "^https:\\/\\/www\\.bing\\.com\\/fd\\/ls\\/GLinkPingPost\\.aspx.*acli?c?k",
-            ),
-        ),
-    )
+    internal val providerList = convertJsonToList(remoteSettingsDownloader.createNestedJsonObject())
+
+    private fun convertJsonToList(jsonObject: JSONObject): List<SearchProviderModel> {
+        val jsonArray = jsonObject.getJSONArray("data")
+        val providerList = mutableListOf<SearchProviderModel>()
+        for (i in 0 until jsonArray.length()) {
+            val itemObject = jsonArray.getJSONObject(i)
+            val schema = itemObject.getLong("schema")
+            val taggedCodes = itemObject.getJSONArray("taggedCodes").toListOfString()
+            val telemetryId = itemObject.optString("telemetryId")
+            val organicCodes = itemObject.getJSONArray("organicCodes").toListOfString()
+            val followOnParamNames = if (jsonObject.has("followOnParamNames")) {
+                itemObject.getJSONArray("followOnParamNames").toListOfString()
+            } else {
+                null
+            }
+            val codeParamName = itemObject.optString("codeParamName")
+            val queryParamName = itemObject.optString("queryParamName")
+            val followOnCookies = if (jsonObject.has("followOnCookies")) {
+                itemObject.getJSONArray("followOnParamNames").toListOfCookies()
+            } else {
+                null
+            }
+            val searchPageRegexp = itemObject.optString("searchPageRegexp")
+            val expectedOrganicCodes = if (jsonObject.has("expectedOrganicCodes")) {
+                itemObject.getJSONArray("expectedOrganicCodes").toListOfString()
+            } else {
+                null
+            }
+            val extraAdServersRegexps = itemObject.getJSONArray("extraAdServersRegexps").toListOfString()
+            val item = SearchProviderModel(
+                schema,
+                taggedCodes,
+                telemetryId,
+                organicCodes,
+                searchPageRegexp,
+                queryParamName,
+                codeParamName,
+                followOnParamNames,
+                extraAdServersRegexps,
+                followOnCookies,
+                expectedOrganicCodes,
+                extraAdServersRegexps
+            )
```

Comment:
I think we could simplify this while maintaining the nice readability you've got here. We could create the `SearchModelProvider` directly and move to its own function, such as:

```
   private fun JSONObject.toSearchModelProvider(): SearchProviderModel? =
        try {
            val extraAdServersRegexps = getJSONArray("extraAdServersRegexps").toListOfString()
            SearchProviderModel(
                schema = getLong("schema"),
                taggedCodes = getJSONArray("taggedCodes").toListOfString(),
                telemetryId = optString("telemetryId"),
                organicCodes = getJSONArray("organicCodes").toListOfString(),
                searchPageRegexp = optString("searchPageRegexp"),
                queryParamName = optString("queryParamName"),
                codeParamName = optString("codeParamName"),
                followOnParamNames = if (has("followOnParamNames")) {
                    getJSONArray("followOnParamNames").toListOfString()
                } else {
                    null
                },
                ...
            )
        } catch (e: JSONException) {
            ...
        }
```
We should also try to catch any exceptions that happen with `getJSONArray`

Reply:
then we could do something similar to [this](https://github.com/mozilla-mobile/firefox-android/blob/fa5dd74477bcc830808c0a9991504aee75a2cd63/android-components/components/browser/icons/src/main/java/mozilla/components/browser/icons/extension/IconMessage.kt#L77)